### ISSUE TITLE
Neptune 4/4 Pro/Sovol SV08 Config Files Missing

### ISSUE THREAD

**Issue by linhungsam** [coprint/configs#1]
Hi, 

FYI, the default config printer.cfg here as well as other files may not work for other printers like Neptune 4/4 Pro/SV08. Do you have specific files for the specific printer like you have for Creality?

For example, these settings in your printer.cfg are not applicable for SV08

[bed_screws]
screw1: 10, 10
screw2: 195, 10
screw3: 195, 198
screw4: 10, 198

The PDF configuration instruction is only for Creality, may be add some notes so the user are aware of the configuration. Misconfiguration may cause damage to customer printer.

Sam

**Comment by 05wrxsti** [comment]
Kindly add Neptune 4 Plus/Max configs to the list as well

**Comment by xTremePower** [comment]
+1 ..
I mounted all chroma KCM + ECM to my Neptune 4 Max ...
Now I have to figure out how to modify all flipper configs to make it work myself .... 
Because even if they promised printer compatibility ... It's impossible to use it now ... Still have to wait ... 2 years of late is not enought ....

**Comment by linhungsam** [comment]
> +1 .. I mounted all chroma KCM + ECM to my Neptune 4 Max ... Now I have to figure out how to modify all flipper configs to make it work myself .... Because even if they promised printer compatibility ... It's impossible to use it now ... Still have to wait ... 2 years of late is not enought ....

my N4 Pro now works with ChromaSet. I have to modify the macros as well as nozzle offset because stock Neptune nozzle is in the middle while ChromaHead is on the left side. If you dont want to wait for CoPrint, you can try yourself, but make sure your calculation is right or ChromaHead may crash into bed plate when home or doing probe calibration.

I learned a lot from setting ChromaSet myself.

Good Luck and happy printing

**Comment by xTremePower** [comment]
> > +1 .. I mounted all chroma KCM + ECM to my Neptune 4 Max ... Now I have to figure out how to modify all flipper configs to make it work myself .... Because even if they promised printer compatibility ... It's impossible to use it now ... Still have to wait ... 2 years of late is not enought ....
> 
> my N4 Pro now works with ChromaSet. I have to modify the macros as well as nozzle offset because stock Neptune nozzle is in the middle while ChromaHead is on the left side. If you dont want to wait for CoPrint, you can try yourself, but make sure your calculation is right or ChromaHead may crash into bed plate when home or doing probe calibration.
> 
> I learned a lot from setting ChromaSet myself.
> 
> Good Luck and happy printing

Hello, 

In this case, why not just share with others here the files pack to help others ?
I can do it, it's not a problem, my problem is the lack of free time for do it ...

Can you please just share them and all Neptune users can use them, we will just need to change print bed size

**Comment by MythicWebsite** [comment]
Someone sharing the files would be great, I have a Neptune 4 Pro, but absolutely cannot get it to work. The instructions just don't make sense when they're showing you with a different interface and telling you to do steps that you cannot with this printer. I tried to just figure it out, but in the end I could not get there. It refuses to detect the KCM and trying to list the USB devices and changing to either of the two just doesn't work. Now my printer is sitting there unable to be used when it's practically on top of the supported printer list.

**Comment by linhungsam** [comment]
> > > +1 .. I mounted all chroma KCM + ECM to my Neptune 4 Max ... Now I have to figure out how to modify all flipper configs to make it work myself .... Because even if they promised printer compatibility ... It's impossible to use it now ... Still have to wait ... 2 years of late is not enought ....
> > 
> > 
> > my N4 Pro now works with ChromaSet. I have to modify the macros as well as nozzle offset because stock Neptune nozzle is in the middle while ChromaHead is on the left side. If you dont want to wait for CoPrint, you can try yourself, but make sure your calculation is right or ChromaHead may crash into bed plate when home or doing probe calibration.
> > I learned a lot from setting ChromaSet myself.
> > Good Luck and happy printing
> 
> Hello,
> 
> In this case, why not just share with others here the files pack to help others ? I can do it, it's not a problem, my problem is the lack of free time for do it ...
> 
> Can you please just share them and all Neptune users can use them, we will just need to change print bed size

It is not just "change the print bed size" that simple. There are several locations you need to adjust. I will share my configs when they are ready as there are few things i want to fine tune and test, but dont expect them to be available soon as I dont have time to work on it full time. Also my N4 Pro is upgraded with OpenNept4une firmware so its config will be different then the stock N4 Pro

![467183659_10169579135235370_4384335620668959914_n](https://github.com/user-attachments/assets/cdc05b8f-9fe0-4926-b443-f2f3fd039f3b)

![467329723_10169579134950370_4248237395241948983_n](https://github.com/user-attachments/assets/8fbe94c2-bae5-4df9-b116-23dea1717e28)

![467494243_10169579134955370_9163828669919774501_n](https://github.com/user-attachments/assets/b4e7425e-ca70-42a2-a9b4-3dc65a3b9da9)

**Comment by linhungsam** [comment]
> Someone sharing the files would be great, I have a Neptune 4 Pro, but absolutely cannot get it to work. The instructions just don't make sense when they're showing you with a different interface and telling you to do steps that you cannot with this printer. I tried to just figure it out, but in the end I could not get there. It refuses to detect the KCM and trying to list the USB devices and changing to either of the two just doesn't work. Now my printer is sitting there unable to be used when it's practically on top of the supported printer list.

you are right. The instructions for modifying users' printer.cfg are not well prapred, missing some critical sections, and may break some printers. DONT remove/modify these sections:
[mcu]
[stepper_x]
[stepper_y]
[stepper_z]
[heater_bed]
[printer]

and these (they forget to mention):
[tmc2209 stepper_x]
[tmc2209 stepper_y]
[tmc2209 stepper_z]
[verify_heater heater_bed]
etc ...

in summary, those "include" lines are the only lines you should add, leave the rest untouch if you are not sure.

I will share my, but can't guarantee when as I dont have full time to work on this, and I installed custom firmware OpenNept4une for my N4 Pro so stock config may be slightly different

**Comment by MythicWebsite** [comment]
So, I'm still absolutely struggling to get this to work. I've managed to figure out how to get it to connect to the KCM, however now it just gives me a "Internal error during connect: can't convert negative number to unsigned" error.

**Comment by mjfsch** [comment]
> So, I'm still absolutely struggling to get this to work. I've managed to figure out how to get it to connect to the KCM, however now it just gives me a "Internal error during connect: can't convert negative number to unsigned" error.

if your getting this error, it means your klipper version is too low compared to the version installed on the coprint MCU's

as far as i know the only way to get around this is to move to OpenNept4une firmware as they seem to not be publishing updates for your printer to any newer versions
